[CL-3870] Add Hoplr SSO

alexander-cit opened the pull request:
# Changelog
## Added
* [CL-3870] Add Hoplr SSO


[CL-3870]: https://citizenlab.atlassian.net/browse/CL-3870?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

alexander-cit committed:
[CL-3870] Add hoplr SSO engine and basic button

alexander-cit committed:
[CL-3870] Configure and test Hoplr SSO

Commit:
Translations updated by CI (extract-intl)

alexander-cit committed:
[CL-3870] Configure test and production environments

alexander-cit committed:
[CL-3870] Avoid rubocop offence

kogre commented on back/engines/commercial/id_hoplr/app/lib/id_hoplr/hoplr_omniauth.rb:
```diff
@@ -0,0 +1,70 @@
+module IdHoplr
+  class HoplrOmniauth < OmniauthMethods::Base
+    def profile_to_user_attrs(auth)
+      first_name, last_name = auth.info.name.split(' ', 2)
```
That's annoying. Sure they're not passing these separately?

alexander-cit replied:
Not at the moment. Here's what they pass https://github.com/CitizenLabDotCo/citizenlab/pull/5563/files#diff-252e519f913e4a470872602efddf82392cf6730c667a6a80801a25bb54a0964bR14

I can ask them to pass it of course, but I supposed it's not worth it.

kogre reviewed (approved):
Smooth!

alexander-cit commented:
Related to https://github.com/CitizenLabDotCo/cl2-component-library/pull/736

alexander-cit committed:
[CL-3870] Failed attempt to make logout work

alexander-cit committed:
[CL-3870] Update component library version

alexander-cit committed:
[CL-3870] Do not implement RP-initiated logout

Session Management
https://openid.net/specs/openid-connect-session-1_0.html
(that includes RP-initiated logout
https://openid.net/specs/openid-connect-rpinitiated-1_0.html)
is OPTIONAL, so we don't need and maybe should not implement it to
keep it simpler and more reliable.
